backend/src/main/java/gov/cdc/usds/simplereport/api/patientExperience/PatientExperienceController.java

```diff
@@ -0,0 +1,117 @@
+
+  @Autowired
+  private PersonService ps;
+
+  @Autowired
+  private PatientLinkService pls;
+
+  @Autowired
+  private TestOrderService tos;
+
+  @Autowired
+  private TestEventService tes;
+
+  @Value("${feature-flags.patient-links:false}")
+  private boolean patientLinksEnabled;
+
+  /**
+   * Verify that the patient-provided DOB matches the patient on file for the
+   * patient link id. It returns the full patient object if so, otherwise it
+   * throws an exception
+   */
+  @PutMapping("/link/verify")
+  public PxpPersonWrapper getPatientLinkVerify(@RequestBody PxpApiWrapper<Void> body)
+      throws InvalidPatientLinkException {
+    if (!patientLinksEnabled) {
+      throw new FeatureFlagDisabledException("Patient links not enabled");
+    }
+    Person p = pls.getPatientLinkVerify(body.getPlid(), body.getDob());
+    TestEvent te = tes.getLastTestResultsForPatient(p);
+    return new PxpPersonWrapper(p, te);
+  }
+
+  @PutMapping("/patient")
+  public Person updatePatient(@RequestBody PxpApiWrapper<Person> body) throws InvalidPatientLinkException {
+    if (!patientLinksEnabled) {
+      throw new FeatureFlagDisabledException("Patient links not enabled");
+    }
+    pls.getPatientLinkVerify(body.getPlid(), body.getDob()); // gotta verify!
+
+    PatientLink pl = pls.getPatientLink(body.getPlid());
+    UUID facilityId = pl.getTestOrder().getFacility().getInternalId();
+    String patientId = pls.getPatientLinkVerify(body.getPlid(), body.getDob()).getInternalId().toString();
+    Person person = body.getData();
+
+    return ps.updatePatient(facilityId, patientId, parseString(person.getLookupId()),
+        parseString(person.getFirstName()), parseString(person.getMiddleName()), parseString(person.getLastName()),
+        parseString(person.getSuffix()), person.getBirthDate(), parseString(person.getStreet()),
+        parseString(person.getStreetTwo()), parseString(person.getCity()), parseState(person.getState()),
+        parseString(person.getZipCode()), parsePhoneNumber(person.getTelephone()), person.getRole(),
+        parseEmail(person.getEmail()), parseString(person.getCounty()), parseRace(person.getRace()),
+        parseEthnicity(person.getEthnicity()), parseGender(person.getGender()), person.getResidentCongregateSetting(),
+        person.getEmployedInHealthcare());
+  }
+
+  @PutMapping("/questions")
+  public void patientLinkSubmit(@RequestBody PxpApiWrapper<AoEQuestions> body) throws InvalidPatientLinkException {
+    if (!patientLinksEnabled) {
+      throw new FeatureFlagDisabledException("Patient links not enabled");
+    }
+    Person patient = pls.getPatientLinkVerify(body.getPlid(), body.getDob());
+    String patientID = patient.getInternalId().toString();
+
+    AoEQuestions data = body.getData();
+    Map<String, Boolean> symptomsMap = parseSymptoms(data.getSymptoms());
+
+    tos.updateTimeOfTestQuestions(patientID, data.getPregnancy(), symptomsMap, data.isFirstTest(),
```

Comment:
It's kind of bananas that we're converting a string to a UUID in the deserialization layer, then converting it back to a string and then back to a UUID again to go through the service layer. Not necessarily a today fix, since it's got wider scope, but still, 🍌 🍌

Reply:
I just spent a couple of minutes digging into this – because it's bad and we should feel bad about it – but it's definitely more than I want to bite off in this already too-big PR

🍌 🙈